assets/js/src/core/components/split-layout/components/divider/divider.tsx

```diff
@@ -0,0 +1,109 @@
+*/
+
+import React, { useRef, useEffect, useState } from 'react'
+import cn from 'classnames'
+import { Icon } from '@Pimcore/components/icon/icon'
+import { useStyles } from './divider.styles'
+
+interface DividerProps {
+  onMouseResize?: (event: MouseEvent) => void
+  onKeyboardResize?: (event: React.KeyboardEvent<HTMLDivElement>) => void
+}
+
+export const Divider = ({ onMouseResize, onKeyboardResize }: DividerProps): React.JSX.Element => {
+  const dividerRef = useRef<HTMLDivElement>(null)
+  const enableResize = useRef(false)
+
+  const [isHovered, setIsHovered] = useState(false)
+  const [isFocused, setIsFocused] = useState(false)
+  const [isMoving, setIsMoving] = useState(false)
+
+  const { styles } = useStyles()
+
+  useEffect(() => {
+    document.addEventListener('mousemove', handleMouseMove)
+    document.addEventListener('mouseup', handleMouseUp)
+
+    return () => {
+      document.removeEventListener('mousemove', handleMouseMove)
+      document.removeEventListener('mouseup', handleMouseUp)
+    }
+  }, [])
+
+  const isResizable = onMouseResize !== undefined
+
+  // Covered next cases:
+  // 1. Divider is hovered
+  // 2. Divider is focused by keyboard
+  // 3. Mouse is moving out of Divider area
+  const isButtonVisible = isResizable && (isHovered || isFocused || isMoving)
+
+  const handleMouseDown = (): void => { enableResize.current = true }
+
+  const handleMouseUp = (event: MouseEvent): void => {
+    enableResize.current = false
+    setIsMoving(false)
+
+    // Hide resize icon in case if user released click outside divider area
+    if (event.target !== dividerRef.current) {
+      setIsHovered(false)
+    }
+  }
+
+  const handleMouseEnter = (): void => { setIsHovered(true) }
+
+  const handleMouseLeave = (): void => { setIsHovered(false) }
+
+  const handleFocus = (): void => { setIsFocused(true) }
+
+  const handleBlur = (): void => { setIsFocused(false) }
+
+  const handleMouseMove = (event: MouseEvent): void => {
+    if (enableResize.current && onMouseResize !== undefined) {
+      onMouseResize?.(event)
+      setIsMoving(true)
+    }
+  }
+
+  return (
+    <div
+      className={ cn(styles.dividerContainer, { [styles.resizable]: isResizable }) }
+      onMouseDown={ handleMouseDown }
+      onMouseEnter={ handleMouseEnter }
+      onMouseLeave={ handleMouseLeave }
+      ref={ dividerRef }
+      role={ 'button' }
+      tabIndex={ -1 }
+    >
+      <div
+        className={ styles.divider }
+        onBlur={ handleBlur }
+        onFocus={ handleFocus }
+        onKeyDown={ onKeyboardResize }
+        role={ 'button' }
+        tabIndex={ 0 }
+      />
+
+      {isButtonVisible && (
+        <div className={ styles.iconContainer }>
+          <Icon
+            className={ styles.icon }
+            name={ 'chevron-selector-horizontal' }
+            options={ { height: 14, width: 14 } }
+          />
+        </div>
+      )}
```

Comment:
hmm, I'm wondering if we could use our IconButton Component in that case?
Because it already looks pretty similar. 

Maybe we just have to introduce a new type variant, in case we can't match it with the current options.

Reply:
Do we really need an altering on existing styles? 

I mean we also could introduce a new theme or type.
That should not have much effect on existing styles.

![image](https://github.com/user-attachments/assets/b40f3354-734c-4f80-a6f3-a3257ce3ce3a)

And maybe we also should add a new sizing in case we can't match that yet.